tests/test_experimentalist.py

```diff
@@ -0,0 +1,164 @@
+    return tuple(map(random.choice, pools))
+
+
+def gridsearch_pool(ivs: List[IV]):
+    """Returns Cartesian product of sets"""
+    # Get allowed values for each IV
+    l_iv_values = [iv.allowed_values for iv in ivs]
+
+    # Return Cartesian product of all IV values
+    return product(*l_iv_values)
+
+
+def random_sampler(values, n):
+    if isinstance(values, collections.Iterable):
+        values = list(values)
+    random.shuffle(values)
+    samples = values[0:n]
+
+    return samples
+
+
+def weber_filter(values):
+    return filter(lambda s: s[0] >= s[1], values)
+
+
+def uncertainty_sampler(
+    model,
+):
+    model
+    # Theory
+    # n samples to select
+
+
+def test_experimentalist():
+    # %% Load the data
+    # datafile_path = pathlib.Path(__file__).parent.parent.joinpath(
+    #     "example/sklearn/darts/weber_data.csv"
+    # )
+    # data = pd.read_csv(datafile_path)
+
+    # specify independent variables
+    iv1 = IV(
+        name="S1",
+        allowed_values=np.linspace(0, 5, 5),
+        units="intensity",
+        variable_label="Stimulus 1 Intensity",
+    )
+
+    iv2 = IV(
+        name="S2",
+        allowed_values=np.linspace(0, 5, 5),
+        units="intensity",
+        variable_label="Stimulus 2 Intensity",
+    )
+
+    iv3 = IV(
+        name="S3",
+        allowed_values=[0, 1],
+        units="binary",
+        variable_label="Stimulus 3 Binary",
+    )
+
+    # specify dependent variable with type
+    dv1 = DV(
+        name="difference_detected",
+        value_range=(0, 1),
+        units="probability",
+        variable_label="P(difference detected)",
+        type=ValueType.SIGMOID,
+    )
+
+    dv2 = DV(
+        name="difference_detected_sample",
+        value_range=(0, 1),
+        units="response",
+        variable_label="difference detected",
+        type=ValueType.PROBABILITY_SAMPLE,
+    )
+
+    metadata = VariableCollection(
+        independent_variables=[iv1, iv2, iv3],
+        dependent_variables=[dv1, dv2],
+    )
+
+    n_trials = 25
+
+    # Set up pipeline functions with partial
+    pooler_callable = partial(gridsearch_pool, ivs=metadata.independent_variables)
+    sampler = partial(random_sampler, n=n_trials)
+
+    pipeline_random_samp = Pipeline(
+        pooler_callable,
+        weber_filter,
+        sampler,
+    )
+
+    results = pipeline_random_samp.run()
+
+    # ***Checks***
+    # Gridsearch pool is working as expected
+    pool_len = len(list(pipeline_random_samp.pool()))
+    pool_len_expected = np.prod(
+        [len(s.allowed_values) for s in metadata.independent_variables]
+    )
+    assert pool_len == pool_len_expected
+
+    # Is sampling the number of trials we expect
+    assert len(results) == n_trials
+
+    # Filter is selecting where IV1 >= IV2
+    assert all([s[0] >= s[1] for s in results])
+
+    # Is sampling randomly
+    results2 = pipeline_random_samp.run()
+    assert results != results2
```

Comment:
could these in principle be equal, albeit with a vanishingly small probability?

Reply:
Yeah that's true. I thought about that. Just a quick a dirty way to test it as we're developing! I suppose a more robust version with less the probability would be to run it many more times and make sure the results are not all the same.

Reply:
gotcha! quick and dirty to get things in motion certainly makes sense, i just couldn't resist asking :)